### PR TITLE
feat(findings): add replay-backed Okta tampering evaluator

### PR DESCRIPTION
**Pull request by jonathanhaaswriter**
## Summary
- add a replay-backed Okta policy rule lifecycle tampering evaluator that persists normalized findings in Postgres
- expose REST and Connect bootstrap endpoints to evaluate one source runtime and return the persisted findings with the fixed rule spec
- cover the slice with finding service, bootstrap, and Postgres tests plus a local replay demo

## Testing
- make verify
- go test ./internal/findings ./internal/bootstrap ./internal/statestore/postgres
- local replay demo for a policy.rule.update event

### CODE REVIEW

**Comment by jonathanhaaswriter** [comment]
@droid Please perform a deep code review and security review of this PR. Leave inline comments only for high-confidence, actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review this PR after the latest fixes. Focus on whether the prior code review and security findings are resolved, and call out any remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
### Droid sweep status

Latest sweep: `2026-04-29 22:09 UTC`.

- Unresolved non-outdated review threads: `0`
- CI `verify`: `SUCCESS`
- `droid-review`: `not present`

I’ll keep watching for new review findings/check failures and address them as they appear.

**Comment by jonathanhaaswriter** [comment]
Re-reviewed the still-open outdated thread(s) on the latest branch tip. The branch already caps request body size, keeps findings responses from exposing runtime config, and scopes finding fingerprints by tenant/runtime, so no code change was needed. Validation: `go test ./internal/bootstrap ./internal/findings`. Remaining assigned unresolved threads after this audit: none.

**Comment by jonathanhaaswriter** [comment]
Droid sweep audit complete (final pass)

- Re-scanned all live review threads, including outdated ones
- Verified each remaining thread against the current branch tip
- Resolved every actionable thread (or fixed it inline and pushed)
- 0 unresolved review threads across the whole open-PR stack at this time

PR #494 (fix/feedback-main-followup) and PR #495 (feat/droid-creation-workflows) received additional fixes for SSRF (DNS-rebind) and workflow branch-name validation in this pass.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid please harden this PR end-to-end:

1. Re-review the diff for residual bugs, races, edge cases, security gaps, and untested paths.
2. Apply the smallest correct fixes for any high-confidence issues you find.
3. Strengthen test coverage: add regression tests for every prior review finding on this PR plus the new behavior, including failure paths, boundary inputs, concurrency, and not-found / typed-nil interface guards.
4. Run focused tests and `make verify`. If validators fail, fix until green.
5. Commit and push the fixes to this PR's branch (or, if you lack push permission in this workflow, leave concrete inline suggestions and a summary comment).

Do not change the PR scope, do not modify the default branch, and do not merge.

**Comment by jonathanhaaswriter** [comment]
@droid review

File: internal/statestore/postgres/findings.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+		lastObservedAt = firstObservedAt
+	}
+	var stored findingRow
+	if err := s.db.QueryRowContext(ctx, `
+INSERT INTO findings (
+  id, fingerprint, tenant_id, runtime_id, rule_id, title, severity, status, summary,
+  resource_urns_json, event_ids_json, attributes_json, first_observed_at, last_observed_at
+)
+VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10::jsonb, $11::jsonb, $12::jsonb, $13, $14)
+ON CONFLICT (id)
```
[the remark this answers is not in the document]
Comment (jonathanhaaswriter) [reply internal/statestore/postgres/findings.go]: Fixed in 892438731 by aligning `UpsertFinding` with the table uniqueness invariant and using `ON CONFLICT (fingerprint)`.

**Comment by jonathanhaaswriter** [comment]
@droid review

Please re-review the replay-backed Okta tampering evaluator deeply. Findings code is security-critical and easy to silently break:

1. **Rule correctness and false positives/negatives**: verify the policy.rule.update detection accurately fires on real tampering signals and not on benign updates, that disabled-rule events are handled, and that the evaluator does not regress on Okta event variants (different actor types, target shapes, missing fields).
2. **Tenant/runtime isolation**: ensure findings are stored and read strictly within their runtime/tenant scope, that evaluator inputs cannot mix events across runtimes, and that returned findings cannot leak data from other runtimes via shared evidence/URNs.
3. **Determinism and idempotency**: confirm repeated evaluations on the same replay window produce the same persisted findings (stable IDs, fingerprints, severity, evidence ordering) and do not duplicate or churn rows in Postgres.
4. **Replay correctness**: verify the evaluator consumes the replay stream consistent with PR 458’s semantics, handles event_limit boundaries, missing messages, and sparse streams, and fails closed when replay is incomplete.
5. **Persistence and migration safety**: check Postgres writes for proper conflict keys, transaction boundaries, error handling on partial inserts, and that finding schema is forward-compatible with PR 477 (rule catalog), PR 462 (graph evidence), and PR 479 (evidence links).
6. **Error taxonomy and surface**: ensure REST/Connect endpoints classify errors correctly (`InvalidArgument` for bad runtime ids, NotFound for missing runtimes, retriable server errors for transient issues) and never leak Okta payload contents, tokens, or internal stack traces.
7. **Operational failure modes**: review behavior on empty/replay-exhausted streams, malformed events, evaluator panics, partial finding writes, and whether the rule spec itself can be evolved without breaking persisted findings.

**Comment by jonathanhaaswriter** [comment]
Closing this PR — its substantive feature work is already in `main` via the earlier consolidation rollups, and the unresolved factory-droid findings posted on this PR have been addressed in main via the follow-up PRs:

- #497 (`fix(github): drop non-cursor opaque from terminal audit checkpoints`)
- #499 (`fix: address post-merge feedback across pending stacks`)

PR #499 specifically addresses the findings posted on this stack:

- Reject malformed `root_urn` values in `GetEntityNeighborhood` (PR #460)
- Map `refuted` and `superseded` claim statuses to projection deletion (PR #463)
- Tenant-scope `ListClaims` reads (PR #464)
- Per-runtime serialization of `WriteClaims` (PR #474)
- Restore null-array normalization in `graph_summary` (PR #470)
- Surface blank-root errors from `graphLayering` (PR #468)

This branch has diverged ~50k lines from `main` and rebasing it cleanly is no longer practical; closing here keeps the open queue accurate. If anything from this stack still needs to land, please open a fresh PR cut from `main`.
